lib/rex/exploitation/cmdstager/invokewebrequest.rb

```diff
@@ -0,0 +1,46 @@
+# -*- coding: binary -*-
+
+class Rex::Exploitation::CmdStagerInvokeWebRequest < Rex::Exploitation::CmdStagerBase
+
+  def http?
+    true
+  end
+
+  def user_agent
+    /WindowsPowerShell/
+  end
+
+  def generate(opts = {})
+    if opts[:payload_uri].nil?
+      raise "#{self.class.name}##{__callee__} missing opts[:payload_uri]"
+    end
+
+    opts[:temp] ||= '%TEMP%'
```

Comment:
Just confirming if this is meant to mutate the options passed in?

Reply:
The options passed in are also meant to keep state. `CmdStager` is... difficult. So yes, you are correct.